src/components/EventUserTable.js

```diff
@@ -27,15 +40,117 @@ export class EventUserTable extends Component {
     this.getEventTableData(this.props.event.id);
   }
 
+  /* updates stats and the rows in the table
+     faculty, gender, dietary, and year stats are only computed on the initial render of the component
+     # of registered/checkedin etc. is computed every single time this function is called
+  */
   async getEventTableData(eventID) {
+
     const params = new URLSearchParams({
+      eventID: eventID
+    })
+
+    await fetchBackend('/registration/scanEvent?' + params, 'GET')
+      .then(response => response.json())
+      .then(response => {
+        let heardFrom = {};
+        for (let i = 0; i < response.size; ++i) {
+          const temp = response.data[i];
+          if (temp.hasOwnProperty('heardFrom')) {
+            heardFrom[temp.heardFrom] = heardFrom.hasOwnProperty(temp.heardFrom) ? heardFrom[temp.heardFrom] + 1 : 1;
+          }
+        }
+        this.setState({ heardFrom })
+      })
+
+    const eventParams = new URLSearchParams({
       id: eventID
     });
 
-    fetchBackend('/events/getUsers?' + params, 'GET')
+    fetchBackend('/events/getUsers?' + eventParams, 'GET')
       .then(response => response.json())
-      .then(response => {
-        this.setState({ rows: response });
+      .then(async users => {
+        this.registrationNumbers(users)
+        this.notRegistrationNumbers(users);
+      });
+  }
+
+  async registrationNumbers(users) {
+    let registrationObj = {
+      'registered': 0,
+      'checkedIn': 0,
+      'waitlisted': 0,
+      'cancelled': 0
+    }
+    users.forEach(user => {
+      if (user.hasOwnProperty('registrationStatus')) {
+        switch (user.registrationStatus) {
+          case REGISTRATION_STATUS.REGISTERED:
+            ++registrationObj['registered'];
+            break;
+          case REGISTRATION_STATUS.CHECKED_IN:
+            ++registrationObj['checkedIn'];
+            break;
+          case REGISTRATION_STATUS.WAITLISTED:
+            ++registrationObj['waitlisted'];
+            break;
+          case REGISTRATION_STATUS.CANCELLED:
+            ++registrationObj['cancelled'];
+            break;
+          default:
+            return;
+        }
+      }
+    })
```

Comment:
Do you think it's better to do this (replacing faculties) 
`obj.faculties[user.faculty] = obj.faculties[user.faculty] ? obj.faculties[user.faculty] + 1 : 1;`